atomic_reactor/utils/resource_manager.py

```diff
@@ -0,0 +1,132 @@
+"""
+Copyright (c) 2022 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+
+import os
+import subprocess
+import logging
+import json
+from datetime import datetime
+from shlex import quote
+
+ROOT_PATH = 'resource_manager/'
+logger = logging.getLogger(__name__)
+
+
+class ResourceManager:
+    '''
+        Manages build resources
+    '''
+
+    def __init__(self):
+        pass
```

Comment:
because there are no attributes, I've feeling that we are failing in our OOP model (or we don't need OOP at all).

Maybe I'd suggest to do following (also for easier mocking during testing):

```
RemoteHost(username, host, auth)
  * username
  * host
  * auth
  * check_avail()
  * unlock()
  * lock()

HostsPool()
  * hosts - list of RemoteHosts objects
  * load_from_config()
  * find()
```

Reply:
+1 for this structure